Tests/SwiftCurrent_SwiftUITests/SwiftCurrent_SwiftUITests.swift

```diff
@@ -627,4 +627,48 @@ final class SwiftCurrent_SwiftUIConsumerTests: XCTestCase {
 
         wait(for: [expectOnFinish, expectViewLoaded], timeout: TestConstant.timeout)
     }
+
+    func testWorkflowCorrectlyHandlesState() throws {
+        final class Launcher: ObservableObject {
+            var onFinishCalled = false
+            init(workflow: AnyWorkflow,
+                 responder: OrchestrationResponder,
+                 launchArgs: AnyWorkflow.PassedArgs) {
+                if workflow.orchestrationResponder == nil {
+                    workflow.launch(withOrchestrationResponder: responder, passedArgs: launchArgs)
+                }
+            }
+        }
+        struct ModifiedWorkflowViewExplorer<A, W, C>: View, Inspectable {
+            @Binding var isLaunched: Bool
+
+            let inspection = Inspection<Self>()
+            @State var wrapped: W?
+            @State var workflow: AnyWorkflow
+            @State var launchArgs: AnyWorkflow.PassedArgs
+            @State var onFinish = [(AnyWorkflow.PassedArgs) -> Void]()
+            @State var onAbandon = [() -> Void]()
+
+            @StateObject private var model: WorkflowViewModel
+            @StateObject private var launcher: Launcher
+
+            var body: some View { EmptyView() }
+        }
+
+        struct FR1: View, FlowRepresentable {
+            weak var _workflowPointer: AnyFlowRepresentable?
+
+            var body: some View {
+                Button("Proceed") { proceedInWorkflow() }
+            }
+        }
+
+        let workflowView = WorkflowLauncher(isLaunched: .constant(true))
+            .thenProceed(with: WorkflowItem(FR1.self))
+
+        let explorer = unsafeBitCast(workflowView, to: ModifiedWorkflowViewExplorer<Never, Never, FR1>.self)
+
+        // if the unsafeBitCast worked, then everything is a state var.
+        XCTAssertNil(explorer.wrapped)
```

Comment:
This is true, but also, the bit cast could fail if we reorder the parameters, right?  Does it throw, or is explorer nil?

Also, why do we want `explorer.wrapped` to be nil?

Reply:
No, re-ordering of parameters is relatively unimportant unless those parameters are of the exact same type (like `State<String>`. Even that would only matter if we asserted on their contents. You can re-order properties in either struct as much as you want without inherently breaking the bitcast.

Reply:
`explorer.wrapped` is asserted to be nil, so we read from one of the values. The funny thing about unsafe bit casting is that it's unsafe. So it's possible, although highly unlikely, that somebody writes a struct with MORE properties that takes up the same space in memory. If that happens, when we try to read from a property we could underflow or overflow leading to undefined behavior, which is generally bad. This test will at least expose that and probably blow up spectacularly, should that happen.

Reply:
Oh, final answer. If the bit cast fails (which it was when I wrote the test, cause TDD) it fatal errors.